src/main/resources/static/myScript.js

```diff
@@ -0,0 +1,45 @@
+function validateForm() {
+    let mailHolder = document.getElementById("email-address")
+    let mailValidator = document.getElementById("mail-validator")
+    let mailMessage = "Niepoprawny email"
+    const regex = /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+
+    if(!(checkIfMailIsCorrect(mailMessage, mailHolder, mailValidator, regex))){
+        return false;
+    }
+
+    let titleHolder = document.getElementById("title");
+    let titleValidator = document.getElementById("title-validator");
+    let titleMessage = "Tytuł nie może być pusty";
+    if(!(checkIfNotEmpty(titleMessage, titleHolder, titleValidator))){
+        return false;
+    }
+
+    let contentHolder = document.getElementById("content");
+    let contentValidator = document.getElementById("content-validator");
+    let contentMessage = "Treść nie może być pusta";
+    if(!( checkIfNotEmpty(contentMessage, contentHolder, contentValidator))){
+        return false;
+    }
+
+}
+
+function checkIfNotEmpty(message, htmlElementForValidation, htmlElementWithMessage){
+    if(!(htmlElementForValidation.value.length>0)){
+        htmlElementWithMessage.innerHTML = message;
+        return false;
+    }else{
+        htmlElementWithMessage.innerHTML = "";
+        return true;
+    }
+}
+
+function checkIfMailIsCorrect(message, htmlElementForValidation, htmlElementWithMessage, regex){
```

Comment:
Myślę, że jeśli metoda/funkcja jest wywolywana wewnatrz ifa to lepiej unikac w niej efektow ubocznych w postaci przypisania jakiejs wartosci do `innerHTML`

Reply:
tego chyba nie do końca kumam :(

Reply:
Efekt uboczny w przypadku funkcji czy tez metody to cos co zmienia jakiś stan, w tym wypadku jest to `htmlElementWithMessage.innerHTML = message;`. Ogólnie są one nie do uniknięcia w 100%, ale z reguły polecam starać się ich unikać, zwłaszcza w przypadk funkcji, ktore sa wywolywane wewnątrz ifa (tak jak ta jest wywolywana w linijce 7), gdyz z reguly spodziewamy sie prawie zawsze w przyapdku takich funkcji ze cos tylko sprawdzają. Nazwa funkcji też nie wskazuje na to że coś jest zmieniane, mówii tylko że coś jest sprawdzane, co może wprowadzać w błąd.

Reply:
no ale w jaki inny sposob moge wyswietlic komunikat do uzytkownika ze cos jest nietak nic nie zmieniajac?

Reply:
Jedyą opcję jaką widzę jest wstawienie w miejsce:
```
    if(!(checkIfMailIsCorrectAndPutProperMessage(mailMessage, mailHolder, mailValidator, regex))){
        return false;
    }
```
zmienionego body wywoływanej funkcji czyli:
```
    if (!(mailHolder.value.match(regex))) {
        mailValidator.innerHTML = mailMessage;
        return false;
    } else {
        mailValidator.innerHTML = "";
    }
```
I analogicznie dla pozostałych, ma to swoje wady - wpadają dwie powtarzalne linijki kodu

Reply:
Jedyą opcję jaką widzę jest wstawienie w miejsce:
```
    if(!(checkIfMailIsCorrectAndPutProperMessage(mailMessage, mailHolder, mailValidator, regex))){
        return false;
    }
```
zmienionego body wywoływanej funkcji czyli:
```
    if (!(mailHolder.value.match(regex))) {
        mailValidator.innerHTML = mailMessage;
        return false;
    } else {
        mailValidator.innerHTML = "";
    }
```
I analogicznie dla pozostałych, ma to swoje wady - wpadają dwie powtarzalne linijki kodu